src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -893,6 +893,7 @@
 (def CompileRequest
   {(schema/required-key "certname") schema/Str
    (schema/required-key "environment") schema/Str
+   (schema/required-key "versioned_project") schema/Str
```

Comment:
are we stuck with a required "environment" argument? Could we make them both optional and then ensure there is only one? It seems to me like right now you have to have environment and versioned project, but versioned_project alone is not allowed?

Reply:
Whoops! This is just a typo, they should both be optional

Reply:
updated